shared/lib_battery_powerflow.cpp

```diff
@@ -267,20 +267,18 @@ void BatteryPowerFlow::calculateACConnected()
 
 	// Compute total system output and grid power flow
 	P_grid_to_load_ac = P_load_ac - P_pv_to_load_ac - P_batt_to_load_ac - P_fuelcell_to_load_ac;
-	P_gen_ac = P_pv_ac + P_fuelcell_ac + P_inverter_draw_ac - P_system_loss_ac;
```

Comment:
Is this where the improper accounting was fixed with new variables?

Reply:
Yes, I pretty much restored it to what it was before my last changes when I broke the definition of gen while trying to fix the unallowed grid charging bug. So I just restored the original calculation of gen, didn't add any new variables, and refined the error-checking in line 276 so that it should grid-charging preference is enforced.